src/SourceCode.Chasm/TreeNodeMap.cs

```diff
@@ -110,16 +112,68 @@ public TreeNodeMap(ICollection<TreeNode> nodes)
             _nodes = DistinctSort(array, true);
         }
 
-        private TreeNodeMap(ReadOnlyMemory<TreeNode> nodes)
+        private TreeNodeMap(in ReadOnlyMemory<TreeNode> nodes)
         {
             _nodes = nodes;
         }
 
         #endregion
 
-        #region Methods
+        #region Merge
 
-        private static ReadOnlyMemory<TreeNode> Merge(TreeNodeMap first, TreeNodeMap second)
+        public TreeNodeMap Add(in TreeNode node)
```

Comment:
Renamed this method from `Merge` to `Add`, left the other ones as-is (besides shuffling their order)